include/SFML/Audio/Music.hpp

```diff
@@ -244,10 +248,8 @@ class SFML_AUDIO_API Music : public SoundStream
     ////////////////////////////////////////////////////////////
     // Member data
     ////////////////////////////////////////////////////////////
-    InputSoundFile            m_file;     //!< The streamed music file
-    std::vector<std::int16_t> m_samples;  //!< Temporary buffer of samples
-    std::recursive_mutex      m_mutex;    //!< Mutex protecting the data
-    Span<std::uint64_t>       m_loopSpan; //!< Loop Range Specifier
+    struct Impl;
+    const std::unique_ptr<Impl> m_impl; //!< Implementation details
```

Comment:
I'd also love to stop using pimpl but I was under the impression it would require adding mini audio headers to AudioDevice.hpp. How is that not true? We need to know the definitions of `ma_log` and other miniaudio types.

Reply:
We need pimpl sometimes because we want miniaudio to be remain an implementation detail and not leak from public header, but in the case of Music and SoundBuffer (and maybe more), miniaudio is not involved at all (unless I missed something).

Reply:
My bad I was looking at AudioDevice.